lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -56,9 +130,8 @@ public static function get_theme_data( $deprecated = array(), $settings = array(
 
 			if ( $wp_theme->parent() ) {
 				// Get parent theme.json.
-				$parent_theme_json_file = static::get_file_path_from_theme( 'theme.json', true );
-				if ( '' !== $parent_theme_json_file ) {
-					$parent_theme_json_data = static::read_json_file( $parent_theme_json_file );
+				if ( wp_theme_has_theme_json( $wp_theme->parent()->get_stylesheet() ) ) {
```

Comment:
@felixarntz @oandregal Is the perfect example of why we need this function to allow for passing a stylesheet. In this context, we need to know if the parent ( not the current ) theme has theme json. This is need for every request, making this change required.